setup.py

```diff
@@ -6,9 +6,14 @@
 from setuptools.command.test import test as TestCommand
 
 
-if sys.argv[-1] == 'publish':
-    os.system('python setup.py sdist upload')
-    sys.exit()
+def rc_value():
+    with open('version') as fp:
+        val = fp.read().rstrip()
+    return '{}rc0'.format(val)
```

Comment:
This appears to hard-code all release candidates as rc0. Is that intended? There will never be an rc1, rc2, etc.?

Reply:
This is the default that we use for our puppet modules. Can you suggest a code snippet where this is configureable? We can still set a random RC version by pushing it as a tag I think (which will end up as `TRAVIS_TAG` environment variable which in turn is read by the `version()` function.

Reply:
I have recently been using a versioning tool that pulls releases and tags from github, and parses the version numbers out of the most recent tag, then increments that, based on criteria for pre-release, test release, actual release, etc. For example, if the most recent test release is `v0.1.0+1`, a new test build would automatically create `v0.1.0+2` and add that github release to the repo using the github API. The tool we are using is https://github.com/homedepot/flow.

Of course, doing this requires either a tool that can do it, or a script in your repo, and also requires access from your travis pipeline to actually create a release in github.

The only other way that I could suggest would be to use the same strategy you are using here for the value of `version`, which would be to have a file containing the pre-release number in it. That would have to be ensured that it is updated if you do need to have more rc versions, and would need to be ensured to be reset back to rc0 after the version actually gets released.

Reply:
IMO we can still push random rc versions if we want to. Just if we do a release locally it's always rc0. Are you fine with this for now? We can still make this more flexible later, if we need it. For now  I would like to get this merged so we can do a release rather sooner than later.

Reply:
I think this is fine. We've never even really done rc versions and until we come close to 1.0 this isn't likely to be a problem. Since we can use the tags for different rc numbers anyway this is only a problem if people do releases locally on their own laptop, which nobody should be doing.